streamdeck_ui/gui.py

```diff
@@ -208,27 +208,48 @@ def handle_keypress(deck_id: str, key: int, state: bool) -> None:
             keys = keys.strip().replace(" ", "")
             for section in keys.split(","):
                 # Since + and , are used to delimit our section and keys to press,
-                # they need to be substituded with keywords.
+                # they need to be substituted with keywords.
                 section_keys = [_replace_special_keys(key_name) for key_name in section.split("+")]
 
-            # Translate string to enum, or just the string itself if not found
-            section_keys = [getattr(Key, key_name.lower(), key_name) for key_name in section_keys]
-
-            for key_name in section_keys:
-                try:
-                    if key_name == "delay":
-                        time.sleep(0.5)
+                # Translate string to enum, or just the string itself if not found
+                section_keys = [
+                    getattr(Key, key_name.lower(), key_name) for key_name in section_keys
+                ]
+
+                for key_name in section_keys:
+                    if key_name.startswith("delay"):
+                        sleep_time = key_name.split("delay", 1)[1]
+                        if sleep_time:
+                            try:
+                                sleep_time = float(sleep_time)
+                            except Exception:
+                                print(
+                                    f"Not sleeping, could not convert sleep time to float '{sleep_time}'"
```

Comment:
The linter is complianing this line is longer than 100 characters.
I'll accept as-is and fix.